lib/cocoapods/installer/target_installer/pod_target_installer.rb

```diff
@@ -285,6 +285,16 @@ def custom_module_map
         @custom_module_map ||= target.file_accessors.first.module_map
       end
 
+      # The deployment target for the pod target, which is the maximum of all
+      # the deployment targets for the current platform of the target.
+      #
+      # @return [String] The deployment target.
+      #
+      def deployment_target
+        default = Podfile::TargetDefinition::PLATFORM_DEFAULTS[target.platform.name]
+        target.specs.map { |spec| spec.deployment_target(target.platform) || default }.max
```

Comment:
Won't this fail in the standard problem case `ruby -e 'p ["10.10", "10.9"].max'`?